src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,282 @@
+
+        //Represents 'MM'
+        Integer month = Integer.parseInt(dateNumbers[1]);
+
+        //Represents 'dd'
+        Integer day = Integer.parseInt(dayAndTime[0]);
+
+        //Represents 'hh:mm'
+        String time = dayAndTime[1];
+        String[] timeNumbers = time.split(":");
+
+        //Represents 'hh'
+        Integer hour = Integer.parseInt(timeNumbers[0]);
+
+        //Represents 'mm'
+        Integer minutes = Integer.parseInt(timeNumbers[1]);
+
+        return LocalDateTime.of(year, month, day, hour, minutes);
+    }
+
+    /**
+     * Checks if the input is list.
+     * @param string The user input.
+     * @return Returns a boolean.
+     */
+    static boolean isList(String string) {
+        return string.equals("list");
+    }
+
+    /**
+     * Creates the ListCommand.
+     * @return ListCommand.
+     */
+    private static Command parseListCommand() {
+        return new ListCommand();
+    }
+
+    /**
+     * Checks if the input starts with "todo"
+     * @param string The user input.
+     * @return Returns a boolean.
+     */
+    static boolean isToDo(String string) {
+        return string.startsWith("todo");
+    }
+
+    /**
+     * Creates a TaskToDo object and use it as a parameter
+     * to create an AddCommand
+     * @param commandLine The user input.
+     * @return AddCommand.
+     * @throws DukeEmptyToDoException Checks if the input includes
+     * a description.
+     */
+    private static Command parseAddCommandTodo(String commandLine) throws DukeEmptyToDoException {
+        if (commandLine.equals("todo")) {
+            throw new DukeEmptyToDoException();
+        }
+        ToDo task = new ToDo(commandLine.substring(5), false);
+        return new AddCommand(task);
+    }
+
+    /**
+     * Checks if the input starts with "deadline"
+     * @param string The user input.
+     * @return Returns a boolean.
+     */
+    static boolean isDeadline(String string) {
+        return string.startsWith("deadline");
+    }
+
+    /**
+     * Checks if the input starts with "event"
+     * @param string The user input.
+     * @return Returns a boolean.
+     */
+    static boolean isEvent(String string) {
+        return string.startsWith("event");
+    }
+
+    /**
+     * Creates a Deadline or Event object and use it as a parameter
+     * to create an AddCommand.
+     * @param commandLine
+     * @return AddCommand
+     * @throws DukeEmptyDeadlineException Checks if the input for deadline includes a description
+     * @throws DukeEmptyEventException Checks if the input for event includes a description
+     * @throws DukeDeadlineFormatException Checks if the deadline contains a date and time
+     * @throws DukeEventFormatException Checks if event contains a date and time.
+     * @throws DukeDateTimeParseException Checks if the data and time is correct.
+     */
+    private static Command parseAddCommandWithDate(String commandLine) throws DukeEmptyDeadlineException,
```

Comment:
I like how you added comments to different parts of the method. It improves readability.